GPT/App.c

```diff
@@ -0,0 +1,86 @@
+/**********************************************************************
+ *  						INCLUDES
+ *********************************************************************/
+#include "App.h"
+
+/**********************************************************************
+ *  						GLOBAL VARIABLES
+ *********************************************************************/
+ 
+uint16 u16OnPeriod =150 ,u16OffPeriod=350, u16SignalDuration=500 ;
+
+/**********************************************************************
+ *  						LOCAL FUNCTIONS PROTOTYPES 
+ *********************************************************************/
+ 
+void app_PwmParameters (uint16 u16Frequency ,uint16 u16Duty ) ;
+
+/**********************************************************************
+ *  						FUNCTIONS IMPLEMENTATION
+ *********************************************************************/
+ 
+ 
+void app_start (void)
+{
+		
+ uint8 u8DutyPercentage = 30;
+ uint8 u8StateFlag = CLEAR_FLAG;
+ uint8 u8Stage = SET_FLAG;
+ uint8 u8PressFlag = CLEAR_FLAG;
+	
+	if (Button_GetState(BUTTON_1,PULL_UP )== BUTTON_PRESSED)
+	{	      
+	   u8StateFlag=SET_FLAG;
+			app_PwmParameters (2,30);  
+	}   	
+  while(u8StateFlag==SET_FLAG)
+	{	
+		Blink_Start2(BLINKING_GPTM, u16SignalDuration , u16OnPeriod, u16OffPeriod);	
+	
+		if (Button_GetState(BUTTON_1,PULL_UP )== BUTTON_PRESSED)
+		{	      
+			 u8PressFlag=SET_FLAG;
+		}   
+		if ( u8PressFlag==SET_FLAG)
+		{ 
+			u8Stage++;  
+			if (u8Stage == FIRST_STAGE) 
+			{
+				u8DutyPercentage = 30;
```

Comment:
You shouldn't use magic numbers

Reply:
@Hala-Abdelkader  so instead of writing '30' I should make a macro named THIRTY ? how is that of any significance to the code readability ? Also I hope you didn't put much load on the naming conventions cause we put so much effort in the logic of the driver itself and got it working perfectly while others who don't have 'naming violations' with a deficient logic got no comments.

Reply:
-Regarding magic numbers, you can write it **first_blinking_mode** and so on and it will be readable.

-Your effort should be put to make logic correctly with following the naming conventions also as required from you in the task

-You should focus more on your logic not others' logic and if you have seen anyone who didn't follow naming conventions you should tell him that when you review his task and **this is the main reason why you have to review others' delivery.**

-Note: In your video, you didn't do 90% duty cycle in a clear way so, please fix it when you have time.

Reply:
@Hala-Abdelkader We did record another video showcasing the 90% duty cycle in a more clear way and will upload it shortly.
Thanks for your input, I will keep it on mind for the upcoming tasks and as I said I hope the scores aren't affected severely with them 🙏🏻

Reply:
Please mention me when you upload the new video

Reply:
@Hala-Abdelkader  ALL the above comments are now done and handled 👍🏻
You can find them in 'COMMENTS' folder.
Thanks for your time.